### ISSUE TITLE
"No-op" arrow functions typed against Void

### ISSUE THREAD

**Issue by Gama11** [HaxeFoundation/haxe#8128]
```haxe
var noop:Void->Void = () -> {};
```

Instead of a no-op, this actually generates code that returns an empty object literal, which can be quite surprising. See generated JS:

```haxe
var noop = function() {
	return { };
};
```

If typed against a `->Void` return, this should probably just generate `function() {}`?

**Comment by back2dos** [comment]
Strongly related to https://github.com/HaxeFoundation/haxe/issues/7376

**Comment by nadako** [comment]
> If typed against a `->Void` return, this should probably just generate `function() {}`?

I wonder if `({}:Void)` in general should (also) be typed as empty block tho. This makes sense, but might be surprising not to have an error there in cases where it looks like a value (that said you'll still get a "cannot use Void as a value" later).

**Comment by Simn** [comment]
This is difficult to fix because the `return` is inserted as syntax-level.

**Comment by kevinresol** [comment]
Perhaps `EFunction( name : Null<String>, f : Function, arrow : Bool )`?
A breaking change for macros but better break at 4.0

**Comment by nadako** [comment]
> Perhaps `EFunction( name : Null<String>, f : Function, arrow : Bool )`?
> A breaking change for macros but better break at 4.0

_If_ we are to break this, we should do it properly, like:

```haxe
enum LocalFunctionKind {
 FAnonymous;
 FNamed(name:String);
 FArrow;
}
```

**Comment by back2dos** [comment]
I don't see that we have to break anything. Just add an optional `arrow` flag to `Function`, no?

**Comment by RealyUniqueName** [comment]
This is going to be fixed by #8539. 
But the generated code will look like this:
```haxe
var noop = function() {
	return;
};
```
That redundant `return` is a subject for another issue: #6420
